docs/howto/multiple-storage-backends.md

```diff
@@ -0,0 +1,272 @@
+
+This example setup configures lakeFS to manage data across two public cloud providers: AWS and Azure:
+
+```yaml
+blockstores:
+    signing:
+      secret_key: "some-secret"
+    stores:
+        - id: "s3-prod"
+          description: "AWS S3 storage for production data"
+          type: "s3"
+          s3:
+            region: "us-east-1"
+        - id: "azure-analytics"
+          description: "Azure Blob storage for analytics data"
+          type: "azure"
+          azure:
+            storage_account: "analytics-account"
+            storage_access_key: "EXAMPLE45551FSAsVVCXCF"
+
+```
+  </div>
+
+  <div markdown="3" id="hybrid">
+
+This hybrid setup allows lakeFS to manage data across both cloud and on-prem storages.
+```yaml
+blockstores:
+    signing:
+      secret_key: "some-secret"
+    stores:
+        - id: "s3-archive"
+          description: "AWS S3 storage for long-term archival"
+          type: "s3"
+          s3:
+            region: "us-west-2"
+        - id: "minio-fast-access"
+          description: "On-prem MinIO for high-performance workloads"
+          type: "s3"
+          s3:
+            force_path_style: true
+            endpoint: 'http://minio.local'
+            discover_bucket_region: false
+            credentials:
+              access_key_id: "minio_access_key"
+              secret_access_key: "minio_secret_key"
+
+```
+  </div>
+</div>
+
+### Key Considerations
+
+* Unique Blockstore IDs: Each storage backend must have a unique id.
+* Persistence of Blockstore IDs: Once defined, an id must not change.
+* S3 Authentication Handling:
+    * All standard S3 authentication methods are supported.
+    * If static credentials are provided, lakeFS will use them. Otherwise, it will fall back to the AWS credentials chain. 
+      This means that for setups with multiple storages of type `s3`, static credentials are required for all but one.
+
+{: .warning}
+> Changing a storage ID is not supported and may result in unexpected behavior. Ensure IDs remain consistent once configured.
+
+### Upgrading from a single storage backend to Multiple Storage backends
+
+When upgrading from a single storage backend to a multi-store setup, follow these guidelines:
+* Use the new `blockstores` structure, **replacing** the existing `blockstore` configuration. Note that `blockstore` and `blockstores` 
+  configurations are mutually exclusive - lakeFS does not support both simultaneously. 
+* Define all previously available [single-blockstore settings](../reference/configuration.md#blockstore) under their respective storage backends.
+* The `signing.secret_key` is a required setting global to all connected stores.
+* Set `backward_compatible: true` for the existing storage backend to ensure:
+  * Existing repositories continue using the original storage backend.
+  * Newly created repositories default to this backend unless explicitly assigned a different one, to ensure a non-breaking upgrade process. 
```

Comment:
We are re-opening the whole default vs. b/c debate here.  I would much prefer not to say "upgrade" here, given that upgrades might not be possible when using external tools, or for customers whose users do not know their storage IDs.